core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala

```diff
@@ -219,6 +220,9 @@ class HadoopRDD[K, V](
       val bytesReadCallback = if (split.inputSplit.value.isInstanceOf[FileSplit]) {
         SparkHadoopUtil.get.getFSBytesReadOnThreadCallback(
           split.inputSplit.value.asInstanceOf[FileSplit].getPath, jobConf)
+      } else if (split.inputSplit.value.isInstanceOf[CombineFileSplit]) {
+        SparkHadoopUtil.get.getFSBytesReadOnThreadCallback(
```

Comment:
Is it guaranteed that all paths in the `CombineFileSplit` have the same filesystem?

Also one related question after I dug around a bit more. Hadoops `FileSystem.getAllStatistics()` returns a list where you can only distinguish one file system from another via the scheme. What happens if two different `hdfs://` filesystems are being read from within the same thread (for instance if two Hadoop RDD's are coalesced)? Is the assumption that this will never happen?